profile/Client/profileClient.go

```diff
@@ -379,17 +380,37 @@ func AggregateSummary(inputMap map[Profile]*Frequency, Operation string) map[Pro
 
 func convertToJSON(Operation string, data []Profile) {
 	var jsonArray []string
-	jsonByte, _ := json.MarshalIndent(data, " ", "   ")
+	jsonByte, err := json.MarshalIndent(data, " ", "   ")
+	if err != nil {
+		log.Fatal("Cannot marshal JSON", err)
+	}
+
 	//unmarshalling here because it is marshalled two times for some reason
 	if err := json.Unmarshal(jsonByte, &jsonArray); err != nil {
-		fmt.Println("Error parsing JSON array:", err)
+		log.Fatal("Cannot unmarshal JSON", err)
 	}
+
 	if len(jsonArray) > 0 {
-		filepath := "Profile_Summary/"
-		err := os.MkdirAll(filepath, 0600)
-		err = os.WriteFile(filepath+Operation+".json", []byte(jsonArray[0]), 0600)
+		directory := "Profile_Summary/"
+		err := os.MkdirAll(directory, 0700)
```

Comment:
```suggestion
		err := os.MkdirAll(directory, 0600)
```
Why do we need exec permission?

Reply:
Without exec we can't `cd` into that directory, and we also can't create files. Here is the current behavior of `600` permission:

![image](https://github.com/user-attachments/assets/e42641a9-1507-4c4d-988b-d3bb21ef0b58)